### PR TITLE
Fix/result link

### PR DESCRIPTION
**Pull request by vyvydkf628**
-share link style bug fix
-change toast to dialog
-refactor gradation button style
-refactor clipboard function
-add zero score image and text

### CODE REVIEW

File: src/components/page/ResultPage.js [hunk 1]
```diff
@@ -147,21 +147,20 @@ class Result extends React.Component {
...
                 </GradationText>
 
                 <ShareLink />
 
-                <GradationButton color="white" onClick={()=>window.location.href = '/'}>
-                    Start a new quiz
-                </GradationButton>
+                <GradationButton 
+                    onClick={()=>window.location.href = '/'}
+                    text={"Start a new quiz"}/>
```
Comment (minsulee2) [inline src/components/page/ResultPage.js:158]: It seems way better!
text={"Start a new quiz"} />
please make space above at the end of the tag

File: src/components/page/ResultPage.js [hunk 2]
```diff
@@ -147,21 +147,20 @@ class Result extends React.Component {
...
 
-                <GradationButton color="white" onClick={()=>window.location.href = '/'}>
-                    Discuss the quiz with others
-                </GradationButton>
+                <GradationButton 
+                    onClick={()=>window.location.href = '/'}
+                    text={"Discuss the quiz with others"}/>
```
Comment (minsulee2) [inline src/components/page/ResultPage.js:162]: ditto

File: src/components/page/ResultPage.js [hunk 3]
```diff
@@ -147,21 +147,20 @@ class Result extends React.Component {
...
 
-                {/* TODO should fix default array for debugging([1,2]) */}
                 <Answers answerSheet={answers}/>
```
Comment (minsulee2) [inline src/components/page/ResultPage.js:164]: ditto

File: src/components/page/ResultPage.js [hunk 4]
```diff
@@ -147,21 +147,20 @@ class Result extends React.Component {
...
 
                 <Footer/>
```
Comment (minsulee2) [inline src/components/page/ResultPage.js:166]: ditto

File: src/components/ui/Answers.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ class Answers extends React.Component {
             const id = result;
             const selected = this.props.answerSheet[1][index];
             const subject = questionsList[id-1].Subject;
-            const answer = answerList[id-1].Answer;
+            const answer = parseInt(answerList[id-1].Answer);
 
             return (
                 <Grid key={index} >
```
Comment (minsulee2) [inline src/components/ui/Answers.js:52]: no space here at the end

File: src/components/ui/ShareLink.js [hunk 6]
```diff
@@ -52,20 +62,34 @@ class ShareLink extends React.Component {
...
+                <Row onClick={this.handleClickOpen}>
+                    <FaLink size={26}/> 
                 </Row>
+
+                <Dialog
+                    open={this.state.clicked}
+                    onClose={this.handleClose}
+                    aria-labelledby="alert-dialog-title"
+                    aria-describedby="alert-dialog-description"
+                >
```
Comment (minsulee2) [inline src/components/ui/ShareLink.js:74]: Make it up please

File: src/components/ui/ShareLink.js [hunk 7]
```diff
@@ -52,20 +62,34 @@ class ShareLink extends React.Component {
...
+                    <DialogTitle id="alert-dialog-title">
+                        URL copied to clipboard
```
Comment (minsulee2) [inline src/components/ui/ShareLink.js:76]: Could you make it like {"URL copied to clipboard"} ?

File: src/components/ui/ShareLink.js [hunk 8]
```diff
@@ -52,20 +62,34 @@ class ShareLink extends React.Component {
...
+                    <DialogContent>
+                        <DialogContentText id="alert-dialog-description">
+                            {window.location.href}
+                        </DialogContentText>
+                    </DialogContent>
+
+                    <DialogActions>
+                        <CopyToClipboard text={window.location.href}>
+                            <Button onClick={this.handleClose} variant="contained" color="primary">
+                                Ok
```
Comment (minsulee2) [inline src/components/ui/ShareLink.js:88]: ditto

File: src/components/ui/button/GradationButton.js [hunk 9]
```diff
@@ -1,38 +1,59 @@
+import React from 'react'
 import styled from 'styled-components'
 
-// TODO how to make boder-radius??
-const GradationButton = styled.div`
+const LevelText1 = styled.div`
```
Comment (minsulee2) [inline src/components/ui/button/GradationButton.js:4]: What are those LevelText1, 2, 3 for?!

File: src/components/page/ResultPage.js [hunk 10]
```diff
@@ -112,7 +112,7 @@ class Result extends React.Component {
     render() {
         const {title, description} = this.state;
         const answers = this.decodeAnswer(new URLSearchParams(this.props.location.search).get("answers"));
-        const score = answers[1].filter((answer,index) => {return answer==answerList[answers[0][index]-1].Answer}).length;
+        const score = answers[1].filter((answer,index) => {return answer===parseInt(answerList[answers[0][index]-1].Answer)}).length;
         const result = _.sample(resultList[score]);
```
Comment (minsulee2) [inline src/components/page/ResultPage.js:116]: Can you declare them on state in the constructor?

File: src/components/page/ResultPage.js [hunk 11]
```diff
@@ -133,7 +133,7 @@ class Result extends React.Component {
 
                 <GradationText fontSize={2.3} fontWeight={"bold"}>
                     {result.Name}
-                </GradationText >
+                </GradationText>
 
                 Top {result.Top}%
```
Comment (minsulee2) [inline src/components/page/ResultPage.js:138]: It doesn't have any style?!

File: src/components/page/ResultPage.js [hunk 12]
```diff
@@ -110,10 +123,7 @@ class Result extends React.Component {
     }
 
     render() {
-        const {title, description} = this.state;
-        const answers = this.decodeAnswer(new URLSearchParams(this.props.location.search).get("answers"));
-        const score = answers[1].filter((answer,index) => {return answer===parseInt(answerList[answers[0][index]-1].Answer)}).length;
-        const result = _.sample(resultList[score]);
+        const {title, description, answers, score, result} = this.state;
```
Comment (vyvydkf628) [inline src/components/page/ResultPage.js:130]: Do I need to delete them too?

**Pull request merged**